python/_infer.py

```diff
@@ -0,0 +1,249 @@
+from typing import Any
+from ._common import *
+from enum import IntEnum
+from collections.abc import Iterable, Union, Tuple, Mapping, Callable
+import abc
+import queue
+
+
+class Tensor:
+
+    def __init__(self) -> None:
+        # tensor metadata
+        self.data_type: DataType = None
+        self.shape: Iterable[int] = None
+        
+        # buffer attributes..
+        self.buffer_address: int = 0
+        self.byte_size: int = 0
+        self.memory_type: Tuple(MemoryType, int) = (MemoryType.CPU, 0)
```

Comment:
Explicit `DeviceId` over `int` for memory typing?
```suggestion
        self.memory_type: Tuple(MemoryType, DeviceID) = (MemoryType.CPU, 0)
```